compiler_gym/envs/llvm/service/ObservationSpaces.cc

```diff
@@ -90,6 +94,70 @@ std::vector<ObservationSpace> getLlvmObservationSpaceList() {
             defaultValue.begin(), defaultValue.end()};
         break;
       }
+      case LlvmObservationSpace::IR2VEC_FA: {
+        ScalarRange featureSize;
+        featureSize.mutable_min()->set_value(0.0);
+        std::vector<ScalarRange> featureSizes;
+        featureSizes.reserve(kIR2VecFeatureDim);
+        for (size_t i = 0; i < kIR2VecFeatureDim; ++i) {
+          featureSizes.push_back(featureSize);
+        }
+        *space.mutable_double_range_list()->mutable_range() = {featureSizes.begin(),
+                                                               featureSizes.end()};
+        space.set_deterministic(true);
+        space.set_platform_dependent(false);
+        std::vector<double> defaultValue(kIR2VecFeatureDim, 0.0);
+        *space.mutable_default_value()->mutable_double_list()->mutable_value() = {
+            defaultValue.begin(), defaultValue.end()};
+        break;
+      }
+      case LlvmObservationSpace::IR2VEC_SYM: {
+        ScalarRange featureSize;
+        featureSize.mutable_min()->set_value(0.0);
+        std::vector<ScalarRange> featureSizes;
+        featureSizes.reserve(kIR2VecFeatureDim);
+        for (size_t i = 0; i < kIR2VecFeatureDim; ++i) {
+          featureSizes.push_back(featureSize);
+        }
+        *space.mutable_double_range_list()->mutable_range() = {featureSizes.begin(),
+                                                               featureSizes.end()};
+        space.set_deterministic(true);
+        space.set_platform_dependent(false);
+        std::vector<double> defaultValue(kIR2VecFeatureDim, 0.0);
+        *space.mutable_default_value()->mutable_double_list()->mutable_value() = {
+            defaultValue.begin(), defaultValue.end()};
+        break;
+      }
+      case LlvmObservationSpace::IR2VEC_FUN_FA: {
+        space.set_opaque_data_format("json://");
+        space.mutable_string_size_range()->mutable_min()->set_value(0.0);
+        space.set_deterministic(true);
+        space.set_platform_dependent(false);
+        std::vector<double> defaultEmbs;
+        for (double i = 0; i < 300; i++) defaultEmbs.push_back(i);
```

Comment:
Please always use braces around `for` loop body, like in lines 119-121.

Reply:
Done!